src/core/src/preprocess/preprocess_steps_impl.cpp

```diff
@@ -130,6 +130,64 @@ void PreStepsList::add_mean_impl(const std::vector<float>& values) {
         "mean " + vector_to_string(values));
 }
 
+void PreStepsList::add_pad_impl(const std::vector<int>& pads_begin,
+                                const std::vector<int>& pads_end,
+                                const std::vector<float>& values,
+                                PaddingMode mode) {
+    std::string name;
+    name = "pad(begin " + vector_to_string(pads_begin) + 
+            ", end " + vector_to_string(pads_end);
+    switch (mode) {
+        case PaddingMode::PAD_CONSTANT:
+            name += ", with " + vector_to_string(values) + ")";
+            break;
+        case PaddingMode::PAD_EDGE:
+            name += ", copied from edge";
+            break;
+        case PaddingMode::PAD_REFLECT:
+            name += ", reflected from tensor";
+            break;
+        case PaddingMode::PAD_SYMMETRIC:
+            name += ", symmetrically added from tensor";
+            break;
+    }
+
+    m_actions.emplace_back(
+        [pads_begin, pads_end, values, mode](const std::vector<Output<Node>>& nodes,
+                                             const std::shared_ptr<Model>& function,
+                                             PreprocessingContext& ctxt) {
+            OPENVINO_ASSERT(!nodes.empty(), "Internal error: Can't add pad for empty input.");
```

Comment:
Can be removed as below there is check if `nodes.size()== 1` so will be not empty.